app/src/main/java/org/simple/clinic/summary/PatientSummaryScreen.kt

```diff
@@ -387,7 +397,48 @@ class PatientSummaryScreen(
   }
 
   override fun contactDoctor(patientTeleconsultationInfo: PatientTeleconsultationInfo, teleconsultationPhoneNumber: String) {
-    // TODO (SM): Build WhatsApp message and start implicit intent.
+    val phoneNumber = teleconsultationPhoneNumber.replace("+", "")
+    val message = StringBuilder("*${patientTeleconsultationInfo.facility.name}* teleconsult request for:")
+        .appendln("")
+        .appendln("")
+        .appendln("*Patient record*:")
+        .appendln("https://app.simple.org/patient/${patientTeleconsultationInfo.patientUuid}")
+        .appendln("")
+
+    if (patientTeleconsultationInfo.bloodPressures.isNotEmpty()) {
+      val bloodPressures = patientTeleconsultationInfo
+          .bloodPressures.joinToString(separator = "\n") {
+            val bpRecordedAtDate = dateFormatter.format(it.recordedAt.toLocalDateAtZone(userClock.zone))
+            "${it.reading.systolic}/${it.reading.diastolic} ($bpRecordedAtDate)"
+          }
+      val bloodPressuresSize = patientTeleconsultationInfo.bloodPressures.size
+      val bloodPressuresTitle = context.resources
+          .getQuantityString(R.plurals.patientsummary_contact_doctor_patient_info_bps, bloodPressuresSize, bloodPressuresSize.toString())
+
+      message.appendln(bloodPressuresTitle)
+          .appendln(bloodPressures)
+          .appendln("")
+    }
+
+    if (patientTeleconsultationInfo.prescriptions.isNotEmpty()) {
+      val medicines = patientTeleconsultationInfo
+          .prescriptions.joinToString(separator = "\n") { "${it.name} ${it.dosage}" }
+      message.appendln("*Current medicines*:")
+          .appendln(medicines)
+          .appendln("")
+    }
+
+    if (patientTeleconsultationInfo.bpPassport.isNullOrBlank().not()) {
+      message.appendln("*BP Passport*: ${patientTeleconsultationInfo.bpPassport}")
+          .appendln("")
+    }
```

Comment:
With multi-line strings it seems much more complex because I have to perform the conditionals inside it (those require me to provide an else condition always), so this seemed like a good option which is easier to read.

Reply:
Alright, let's just extract the building of this message to a separate class for now then.